api/api.go

```diff
@@ -256,11 +267,95 @@ func CreateNamespace(ctx context.Context, logger *logger.Logger, name string, or
 		return nil, err
 	}
 
-	namespace, err := createNamespaceByID(ctx, logger, name, organizationID)
+	namespace, err := createNamespaceWithOwnerID(ctx, logger, name, organizationID)
 
 	if err != nil {
 		return nil, err
 	}
 
 	return namespace, err
 }
+
+func getNamespace(ctx context.Context, logger *logger.Logger, name string) (string, error) {
+	var response struct {
+		RegistryNamespace struct {
+			ID string
+		}
+	}
+
+	query := `
+				query($name: String!) {
+					registryNamespace(
+						name: $name
+					){
+						id
+					}
+			 }`
+	request := client.NewAuthorizedRequest(viper.GetString("token"), query)
+	request.Var("name", name)
+
+	graphQLclient := client.NewClient(viper.GetString("endpoint"), logger)
+
+	err := graphQLclient.Run(ctx, request, &response)
+
+	if err != nil || response.RegistryNamespace.ID == "" {
+		err = fmt.Errorf("Unable to find namespace %s", name)
+	}
+
+	return response.RegistryNamespace.ID, err
+}
+
+func createOrbWithNsID(ctx context.Context, logger *logger.Logger, name string, namespaceID string) (*CreateOrbResponse, error) {
+	var response struct {
+		CreateOrb struct {
+			CreateOrbResponse
+		}
+	}
+
+	query := `mutation($name: String!, $registryNamespaceId: UUID!){
+				createOrb(
+					name: $name,
+					registryNamespaceId: $registryNamespaceId
+				){
+				    orb {
+				      createdAt
+				      id
+				    }
+				    errors {
+				      message
+				      type
+				    }
+				}
+}`
+
+	request := client.NewAuthorizedRequest(viper.GetString("token"), query)
+	request.Var("name", name)
+	request.Var("registryNamespaceId", namespaceID)
+
+	graphQLclient := client.NewClient(viper.GetString("endpoint"), logger)
+
+	err := graphQLclient.Run(ctx, request, &response)
+
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("Unable to create orb %s for namespaceID %s", name, namespaceID))
+	}
+
+	return &response.CreateOrb.CreateOrbResponse, err
+}
+
+// CreateOrb creates (reserves) an orb within a namespace
+func CreateOrb(ctx context.Context, logger *logger.Logger, name string, namespace string) (*CreateOrbResponse, error) {
+	namespaceID, err := getNamespace(ctx, logger, namespace)
+
+	if err != nil {
+		return nil, err
+	}
+
+	orb, err := createOrbWithNsID(ctx, logger, name, namespaceID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return orb, err
```

Comment:
If you `return createOrbWithNsID(ctx, logger, name, namespaceID)` then you can delete the next 5 lines of code.